### PR TITLE
doc(website): Proper `backend_options` documentation

### PR DESCRIPTION
**Pull request by candiduslynx**
Closes https://github.com/cloudquery/cloudquery/issues/14725

### CODE REVIEW

File: website/pages/docs/plugins/sources/homebrew/_configuration.md [hunk 1]
```diff
@@ -8,8 +8,8 @@ spec:
   tables: ["*"]
   destinations:
     - "DESTINATION_NAME"
+  backend_options:
+    table_name: "test_state_table"
```
Comment (hermanschaaf) [inline website/pages/docs/plugins/sources/homebrew/_configuration.md:12]: Bit of a nit, but can we maybe call it something other than `test_state_table` (here and in the other configs). Maybe `homebrew_state_table`?

**Review by erezrokah — APPROVED** [review]

File: website/pages/docs/plugins/sources/homebrew/_configuration.md (hunk 1 above)
Comment (candiduslynx) [reply website/pages/docs/plugins/sources/homebrew/_configuration.md:12]: we have only a single place where we did this, so I left others as is

Comment (candiduslynx) [reply website/pages/docs/plugins/sources/homebrew/_configuration.md]: Renamed to `cq_state_<plugin>` in 036cfc5

Comment (hermanschaaf) [reply website/pages/docs/plugins/sources/homebrew/_configuration.md]: Ah yeah I see, thanks for the fix!

**Pull request merged**